### DISCUSSION TITLE
Making Budgie 10 Fly Again

### DISCUSSION THREAD

**Discussion by JoshStrobl** [BuddiesOfBudgie/budgie-desktop#2] [Ideas]
This is an open discussion on where to take Budgie 10 series while we get to work on Budgie 11. As many of you know, Budgie 10 series was more-or-less frozen for maintenance, and I think we all want to open it back up again.

Here are *just* my ideas. These are not the *sole* ideas. I want **your** ideas.

- Ensure folks are properly enabled on the [new Transifex](https://www.transifex.com/buddiesofbudgie/).
- Set up GitHub Actions or similar systems for CI.
- Implement some form of dconf / gsettings conversion from `/com/solus-project/` to `/org/buddiesofbudgie/`
- Sit down with Ubuntu Budgie folks and figure out what applets should be "mainlined" into Budgie. For example, improving the UX of Budgie Window Shuffler. Improvements from DriveBy into our Drive applet would also be welcome.
- Raven API! No waiting for Budgie 10 series.
- New Internal Theme (I have some design ideas).
- Transform "Dark" toggle to not just apply to any internal theming and be a bit jank overall, but dynamically swap between supported variants of widget themes (e.g. Plata-Lumine to Plata-Noir). Support XDG Standard for dark theme.

**Comment by EbonJaeger** [comment]
My big pain point with Budgie is multi-monitor panel support. If something is in full-screen mode on the main screen, you're kinda stuck because no other screen will have a panel. It will probably require further discussion on the best way to improve this, if desired.

**Reply by JoshStrobl** [reply]
Oh yea not sure how I forgot to add that! I'd love PanelManager to be rewritten to support multiple panels across **multiple** monitors. They don't necessarily have to be something like mirrors (I believe Windows basically mirrors a single panel across monitors), but anything so I could just access Budgie Menu on another display while gaming :D

**Reply by EbonJaeger** [reply]
Yesss! That sounds wonderful! Even if other monitors just have a panel that has an instance of menu, tasklist, and clock, that'd be great. Ideally I suppose it would be nice to configure what applets you want on other monitors, but small steps first.

**Reply by serebit** [reply]
Windows mirrors a single panel across monitors EXCEPT for the system tray.

**Reply by fossfreedom** [reply]
I've got some draft multimonitor - multiple panel code in a test branch here https://github.com/ubuntubudgie/budgie-desktop/tree/multipanel -  I need to merge in current git master; when done might be useful as something we can all reference.

**Reply by rbreaves** [reply]
Also glad to be seeing others mention multi-monitor support for the panel area. Has been something I have also felt like was a bit of an oversight as someone who has used xfce and mate that had that support.

Of course Global Menu users on Ubuntu Budgie are wanting this so that Vala-AppMenu will apply to that multi-monitor panel. And thank you for posting that test branch @fossfreedom I will definitely test that out!

**Comment by guillotjulien** [comment]
Dreaming about some more flexibility for window positioning (and tilling). e.g. On Windows your screen can basically be cut in 4 parts, whereas on Budgie I can only use the left or the right side of the screen (or the whole screen). This is especially useful when you have a big screen and the need to compare stuff (IDEs, spreadsheets, etc).

**Reply by JoshStrobl** [reply]
Yea a lot of this is due to Mutter being more limited. I'm not sure how Pop_OS handles it with their desktop, but tiling certainly can be improved.

**Reply by codic12** [reply]
I believe pop os simply uses a gnome-shell extension (which has some level of integration with mutter) to move things around.

Related to window positioning and tiling, automatic tiling as an option would be *really* nice (and after learning more about the codebase, I'd be happy to help, as I've implemented the same feature in my own window manager) but I'm not sure if in the first place it's in-scope for Budgie. I'm not sure how limited Mutter's API is, but I think it should be doable, considering that even external programs can do it via raw X11 move/resize calls; I'd prefer to just use a Mutter API for that though...

Also I think that if it is possible to simply disable window snapping in Mutter, we can implement it ourselves; we need to simply check x and y position when a window is moved, and if it falls within a range and has been held down for enough time, we draw a preview window until ButtonRelease, then we destroy the preview window.

**Reply by Solarunit** [reply]
I hope it would be possible to disable 4 corners snapping, because 2 side snapping is enough for me :)

**Reply by marcocolnaghi** [reply]
you could try windows shuffler panel plugin, it makes tiling really easy

**Comment by serebit** [comment]
Currently we only have one `panel.ini` file—figuring out sane defaults aside from this and allowing them to be selected via a Meson compiler variable would be cool. That way, we could give an Ubuntu Budgie like preset that would require less effort on their part to customize.

**Reply by JoshStrobl** [reply]
Great idea!

**Comment by Walrusz** [comment]
The feature I'd like to see the most in Budgie 10 is the ability to make the Budgie panel a custom solid colour with a colour picker. I would love to be able to set my panel a solid black colour, as the look some GTK themes give the panel doesn't match the rest of the theme, so it would be great to be able to overwrite that.

**Reply by serebit** [reply]
That's a really cool idea and I don't think it would be too difficult to implement either.

**Reply by Solarunit** [reply]
good idea! And also maybe accent highlight color and separators color?

**Comment by clumsyfingers** [comment]
if i was to wish for one thing it would be window rules, as in app1 always opening in a certain position (and size) and app2 in another, like in kwin and compiz, and devilspie. it seems like its difficult to achieve under wayland though, although i think it works with kwin now. most seem to cater for users with small screens but some of us have huge ones where it makes sense to have windows not overlapping. tiling isnt a really good solution imho

**Reply by JoshStrobl** [reply]
Well I'm not sure how much of that would be in the scope of Budgie 10 series, especially given the limitations of X11. I think once / if we get any form of tiling implemented we'll be able to handle positioning much better for that.

**Reply by fossfreedom** [reply]
It's a really complicated area - UB team member Jacob who developed Window Shuffler is an expert here of all the intracasies involved.  And yes, wayland with its emphasis on window security makes any solution very difficult. Window Shuffler has the ability to define rules, set window sizes etc for initial app opening.  But it will not work for wayland desktop environments ... its an X solution only.

**Reply by JoshStrobl** [reply]
@fossfreedom Yea this is something we'll need to address at the compositor / WM level for Budgie 11.

**Comment by rbreaves** [comment]
To mention this here, I think it would be good to look at how we can offer something akin to "Displays have separate workspaces" as a 1st class feature, not sure if the implementation of it differs when dealing with x11 vs wayland, but this feature would be highly useful.

Currently if you switch workspaces it changes the workspace on all monitors, but when Displays have their separate workspaces then it only changes the workspace on the monitor with the mouse cursor on it. Very helpful when you use other monitors for reference and research most of the time while focusing your efforts on a particular monitor. (Also tends to feel very natural switching workspaces with good touchpad gestures via touchegg.)

I believe the Enlightenment DE/Window manager supported it as well as i3, awesomeWM and a few other tiling managers, but no mainstream DE like Gnome, KDE, xfce, or mate support implementing workspaces in this way.

This is how mac users would enable it.
![Workspaces](https://i.stack.imgur.com/as6cP.png)

Windows users have never had this feature, but it was part of an Insider build, so clearly something Microsoft has been thinking about or trying to implement as well.
![Workspaces](https://winaero.com/blog/wp-content/uploads/2021/02/Virtual-Desktop-per-Display.png)
More details here on the Windows implementation that has still not been merged into Windows 11 either.
https://winaero.com/windows-10-will-soon-have-individual-virtual-desktops-per-display/

**Reply by Solarunit** [reply]
I think there should be this kind of setting. I would  like to have virtual desktops only on Primary display and use the second display as a Standalone monitor.

**Reply by Solarunit** [reply]
Just tested on Fedora Gnome 36 (wayland) 

- different scaling (100/200%) for 2 Monitors works!
- you can switch workspaces only on one of the screen (Primary display) other display is not affected. (Gnome 42 Multitasking setting)


Fedora Gnome 36 (Xorg) 

- different scaling (100/200%) for 2 Monitors doesn't work, selected scaling applies to the both monitors
- you can switch workspaces only on one of the screen (Primary display) other display is not affected.

**Comment by EbonJaeger** [comment]
I mentioned this briefly in the Matrix channel last night, but it would be nice to have broader input:

It would be great to be able to have notification counts and even progress indicators for the tasklist applets. It looks like Cosmic and Dash-to-Dock do this by leveraging UnityIndicator and [subscribing](https://github.com/micheleg/dash-to-dock/blob/39e5597da42124a23d3f0d5dbca69fea1f3fa96b/launcherAPI.js#L16-L24) to the `com.canonical.Unity.LauncherEntry.Update` DBus interface to get count and progress updates.

I'm not sure yet what it would involve to hook something like that up with the current implementation in [Icon]TaskList.

*EDIT: Wiki page on the Unity API: https://wiki.ubuntu.com/Unity/LauncherAPI

**Reply by EbonJaeger** [reply]
After playing around a bit with DBus in Abomination, I think this would work if I knew what provided the messages over DBus. And, I suppose, how to accurately check that I did it right. So far I haven't figured that out. I'm afraid it might be another dependency, and I don't know that we really want to do that.

**Comment by Solarunit** [comment]
Here are some ideas:

- Smooth animation when reordering windows in Task List applet
- Setting for middle click for Task List applet: open a new window or close the window
- Option to disable scrolling for the applets who use that (e.g. Icon Task List)
- Autostart apps on certain workspaces
- Different wallpapers on workspaces
- Alt+F2 search accuracy
- Group/stack similar notifications and button/toggle to quickly unstack notifications
- Option to only manually close some selected (important) notification
- Notifications disappearance timer maybe
- History of all notifications
- Option to hide items in the Power applet, e.g. Suspend, Hibernate

**Reply by JoshStrobl** [reply]
> Smooth animation when reordering windows in Task List applet

Will need to take a look at what functionality is exposed on the Wnck.Tasklist. Would be easier to just add label support to the IconTasklist and rename that, to be honest.

> Group/stack similar notifications and button/toggle to quickly unstack notifications

Probably not going to happen because it is intentionally a Notification Group. Intentional design decision on that front :)

> Autostart apps on certain workspaces

I'm just considering what that'd look like from the perspective of it being exposed in Budgie Desktop Settings. If we can come up with a good UX for it, it's worth considering. Otherwise might need to wait until we have a dedicated control panel.

> Option to hide items in the Power applet, e.g. Suspend, Hibernate

Worth considering once I redesign Budgie Menu ;)

**Comment by kp2401075** [comment]
Here is some more ideas:

- icon task list icons scaling according to panel size.
- custom spacing config for icon task list.

I would also like to help with testing if possible.

**Reply by Solarunit** [reply]
custom spacing option would be nice

**Reply by JoshStrobl** [reply]
Yea we can see about refining the IconTasklist to prioritize higher / next highest step icon size and downscale those, should allow for a greater variety of sizing for the panels without the IconTasklist button iconography being smaller than expected (they follow logical step sizing like 24x24, 32x32, 48x48, etc... at the moment)

**Comment by clumsyfingers** [comment]
further down the line, when you run out of things to build or improve (if ever) id love an rss applet with visual and audible alerts when new entries, perhaps with the entries in raven. ive felt completely handicapped since feedindicator died

**Reply by JoshStrobl** [reply]
Well, once there's a Raven API, who's to say we have to build it? :wink:

**Reply by rbreaves** [reply]
Could probably just as easily build it already w/ Python & appindicators.

**Comment by ringo32** [comment]
got not really a idea... Also WM's are lately also for some grown a much.. besided the looks and ui's..

How is the keyboard workflow.?

**Reply by JoshStrobl** [reply]
We have a wide variety of keyboard bindings as well as inherit many from GNOME via their gnome-settings-daemon work. If you have any concrete suggestions, would be good to hear.

**Reply by ringo32** [reply]
actually probably need to DE hop later on :) Most wwhat i like does it general already have.. have to see what can be something if it standard can tiling is always a good thing of a desktop. without a extention.. (because gnome can tile to but extentions external is different)

**Reply by ringo32** [reply]
also something like a auto generated cheatlist. for beginners you setup at once but in a keyflip sort cheatlist :) just a idea.

**Reply by JoshStrobl** [reply]
@ringo32 I intend on designing a Budgie site that will have documentation, including with keyboard shortcuts, so stay tuned for that :)

**Reply by Solarunit** [reply]
Josh, don't forget `Super+C` :wink:

**Reply by ringo32** [reply]
Ok i am not really into gnome realm but interest trigged 😇

**Reply by rbreaves** [reply]
Distro and DE shortcuts are very subjective things imo. I created a spreadsheet over a year ago that documented many of them thoroughly, but mostly so I could more easily figure out what always remained the same and what didn't for my own projects remapping efforts.

At this point if changes or major changes were to happen I would have to also document the version of Budgie so that I can ensure my remaps do not break. I really don't have any suggestions or complaints for changing keymaps for this reason, it would make maintaining my [kinto.sh](https://github.com/rbreaves/kinto) project a little harder, but if it happens then I will deal with it.

**Comment by rbreaves** [comment]
What about improving mixed low and hidpi monitor setups? I feel like this could be simplified greatly and without any extreme effort really. 

The biggest issue might come from the lack of persistence from xrandr, and imo the best kind of scaling is the sort where you do not have weird odds and ends not scaling properly. This method that I explain in [this reddit comment](https://www.reddit.com/r/UsabilityPorn/comments/ryo099/1920x1080_monitor_w_2880x1800_macbook_perfectly/hrq07cn/) is fairly formulaic, anyone could easily create 4 to 10 levels of scaling and apply the math all the way down from the main hiDPI monitor to the low dpi monitor and then apply it properly based on expected positioning as well.

This would also be a bit of a head turner too because as far as I know.. there is not a single Linux DE out there that actually helps users to apply proper scaling to this degree w/ mixed monitor setups, and so many people just refer to the [HiDPI documentation](https://wiki.archlinux.org/title/HiDPI) from the arch wiki.. which is great, but more often then not will end up with a mess if they are trying to do this with mixed monitors at all.

![1920-w-2880](https://user-images.githubusercontent.com/10969616/148627553-3db790cf-1877-4f17-9697-1368d1dc09b3.jpg)
https://www.reddit.com/r/UsabilityPorn/comments/ryo099/1920x1080_monitor_w_2880x1800_macbook_perfectly/hrq07cn/

I feel like this is what Apple does more or less when a user plugs in an external low dpi monitor - they create a framebuffer either together or individually that makes sense for the desired scale the user had selected previously. All specific references to the actual resolutions are hidden I think in part for this reason because they simply upscale by a full integer and then downscale fractionally. It seems rather obvious and there isn't much reason that I can ascertain on why the last mile on this hasn't been done by any of the DEs under x11 already - to me it isn't a trivial problem & not a difficult one to solve gracefully either.

**Reply by rbreaves** [reply]
I also made this post about a single monitor/laptop that imo sits between high and low dpi given the monitor size, and I used acpi to re-apply the xrandr but was never totally happy with it because I felt like it was not reliable enough on that particular laptop. Not sure why tbh - but sometimes it would not apply or I needed to add delays or even do a full reboot at times.

As long as xrandr settings can be applied and re-applied reliably after sleep and wake cycles I think this method of scaling should be looked at and possibly included in future versions.

https://discourse.ubuntubudgie.org/t/ubuntu-budgie-20-04-fractional-hidpi-for-x11/4777

**Reply by rbreaves** [reply]
BetterScale has now been created. (single monitor support only atm, but will add that later)

https://github.com/rbreaves/betterScale

**Comment by kp2401075** [comment]
I had a feature in mind not sure if this is even possible. but here I go...

being able to select multiple sound output devices in raven.
Use case:
let's say you have 2 Bluetooth speaker connected and you want to output sound trough both of them.

unfortunately my screenshot only shows 1 device, but i hope I'm making some sense.

![image](https://user-images.githubusercontent.com/20717654/148872078-a7915875-1872-41ea-bb21-adf1b9426366.png)

**Comment by glow12121** [comment]
In the User Indicator applet for the bottom panel, I think it makes little sense to play the "Logout" option under a separate pop-up position. It isn't any different or special.

**Comment by clumsyfingers** [comment]
a couple of things that bug me in budgie that should be fixed imho
1: theres an inconsitensy here, in the budgie menu apps are launched when mouse button is released (which it should be, its how all do it afaik) but other places such as in various applets things happen on click (not release) -  the screenshot applet for example, also the app launcher applet

2: the way some menus stay open after click, in particular icon task list right click menu, means i have to do an extra click to activate whatever was launched or it stays unresponsive

3: in gnome and plasma i always quickly quit all apps by middle click in overview. no overview in budgie (hope it comes som day, remember to implement closing app when middle click or similar) so in budgie theres a whole process of rclickin in the icon task list, takes ages. middle click to close in alt+tab dialogue would be a good (temporary) solution

otherwise its near perfect :)

thanks

**Reply by JoshStrobl** [reply]
> theres an inconsitensy here, in the budgie menu apps are launched when mouse button is released (which it should be, its how all do it afaik) but other places such as in various applets things happen on click (not release) - the screenshot applet for example, also the app launcher applet

The reason for this is a technical one. Our Budgie Popover code tracks when mouse press events occur outside the popover. The Popover will automatically hide itself when that is the case, enabling you to effectively click outside the Budgie Menu (as an example) to hide it. Changing the press event to a release event would cause those applets to quickly hide upon click the button again, since it would satisfy the visibility check, and then immediately show again (since it would have been hidden by the popover manager and thereby not visible, and so it would get shown).

So this isn't expected to be changed. For applets where this visibility check isn't needed, for example the behavior is intended to allow you to perform other UX or changes in your environment before closing the popover, this is changed to a release instead of press.

> the way some menus stay open after click, in particular icon task list right click menu, means i have to do an extra click to activate whatever was launched or it stays unresponsive

This is intentional. The idea is to allow for multiple actions to be performed. I appreciate the feedback on it.

> middle click to close in alt+tab dialogue would be a good (temporary) solution

I don't agree with this UX solution. It is not intuitive and mixes mouse and keyboard input. I feel the right click and close all button in the Icon Tasklist is a satisfactory solution at the moment, but I am open to other solutions should they be proposed (whether it be for 10.x or 11).

**Comment by jprostko** [comment]
I'd love to see the return of allowing more than 8 virtual desktops, even if that is going to be pushed to Budgie 11.  I can get by with 8 virtual desktops, but like having 10 or 12 if I can.

**Reply by rbreaves** [reply]
What in the world do you do with 10-12 o.o? I went up to 3-4 awhile back before settling back down to only using 2, but 10-12 sounds nuts.

**Reply by Solarunit** [reply]
I think why not. Especially if you have an extra-wide monitor.

**Reply by jprostko** [reply]
I like to run one application (usually fullscreen) per virtual desktop, and use keyboard shortcuts to switch between the workspaces.  That way I can switch quickly between applications, since I generally put certain applications within the same workspace number.  I usually use a single monitor, and I have gotten used to this workflow from ages ago when I first used BeOS R5 22 years ago.  I think I stuck with 12 workspaces over the years, since BeOS' (and now Haiku's) default workspace switching is ALT+F[1-12].  I would often use CTRL+F[1-12] instead on Linux machines though.  That said, I have somewhat recently switched to using Super+[1-0] now on desktop environments other than Budgie so I can switch between 10 workspaces.  I use Super+[1-8] in Budgie.

It's easy to use up 8 workspaces in my scenario.  For instance, right now, I have a terminal, a primary web browser, a web browser for work-related activities, my Citrix session for work, a Libreoffice Write document, a text editor, and a NoMachine instance open right now.  Yes, that is 7, but if I open up a music player (or anything else) on workspace 7, then I'll be using all my workspaces.  When I run 10 or 12 workspaces, I usually keep the ones at the end available for other things like a messaging app, virtualization software, a second terminal, a graphics program, an IDE, a DAW, etc.

Obviously I can have more than one application per workspace, but I prefer not to, since I like keeping things isolated when I can.  The only time I usually open more than one app on a workspace is if I need to compare side-by-side. That isn't a common thing I do however.

**Reply by JoshStrobl** [reply]
There are no plans on increasing the current limit and this isn't the really the intended usecase for workspaces. I would suggest that you learn how to use basic windowing capabilities such as having more than one window on a workspace at a time, minimizing and maximizing windows, and tiling functionality such as the current side-by-side tiling. Beyond that, you can also have a window be "always on top" of another window.

**Reply by jprostko** [reply]
My use case of workspaces isn't all that atypical, regardless of what you say.  They are meant to either be used like I use them, or else in the oversized desktop scenario where one then pans around.  Spaces in MacOS behave exactly like my use case, and it is common on other desktop environments and operating systems to use them like I use them.  It could become cumbersome by just using a mouse, but when you utilize keyboard navigation between workspaces, and in order to move applications between workspaces as needed, it is quite efficient to use.  Maybe you see no value in having/using so many workspaces, but I'm just stating how I and others may use them.  Even MacOS allows for 16 workspaces, and they tout it as being used for application and context separation.

I remember reading that you stated the change was done to guarantee performance and that was all there was to it.  And if that is the reason, it is definitely a good one.  It hit me by surprise though when the change happened, and the only reason I ever found out about the change was that after the change was made, there was then a crash when a ninth virtual desktop was attempted to be added via the Workspace Switcher.  I reported the issue first on the UB forum, and then later on the bug tracker where you fixed it by making the change more thorough to ensure 8 was the hard cap.  I just ended up recompiling Budgie to use 12 and moved along with my day.

Contrary to what you may think, I do use things like minimize and maximize and close.  Those things don't go away with how I do things, I just don't need to do them often because there often isn't a need to spend energy doing so.  That said, there's times when I move an application from one workspace to another, do what I need to do, and then either minimize it, move it back, or close it.  And I do use side-by-side tiling when I am comparing a terminal to a text document or one document to another, for instance.  (If I had a secondary monitor, I would just use that though.)  Just because I like to use workspaces in the manner I do doesn't mean I throw out all of the other advancements over the years.

I'd argue that relying on always-on-top is pretty lazy, except for certain very limited use cases.  To me, you're probably doing something wrong most of the time if you are relying on functionality like that often.  As for tiling window managers, let's just say I've been down that road and have no interest in going back.  At least in the tiling department, it is fairly counter to how I like to do things, since again, I prefer full screening most applications.  There's also that whole build-it-yourself kind of mentality when it comes to them, and I simply prefer a fully-baked desktop environment like Budgie that stays out of my way and does what it needs to do.

That all said, coming back to what actually matters, if you want or need to keep it at 8, that is fine, and I respect that.  You won't see me obnoxiously asking for it over and over.  I'd appreciate if you'd reconsider for B11 though, especially if any performance problems you saw before get resolved in the process.  In the meantime, on B10, I'll just recompile like before and put the max back at 12, since I never observed any negative effects with my hardware and usage patterns.

**Reply by rbreaves** [reply]
Hmmm.. interesting. I do kinda like the idea of using  Ctrl-Fx to switch workspaces.. not sure if that is bound to anything by default but yea I don't use that combo and makes sense.. Guess I could also just use Ctr-Numkey as well..

**Reply by glow12121** [reply]
> using Ctrl-Fx to switch workspaces

I use Super + 1, 2, 3. It's quite convenient mechanically.

**Reply by Solarunit** [reply]
I use shortcuts on my mouse.

**Comment by ringo32** [comment]
if you see a gnome lot css java json based extentions? there is also a lot people that are pretty handy in bash scripts...its also nice if there people that write in bash also can make some workout in extentions form.. :) or just a simple panel script...

**Reply by JoshStrobl** [reply]
libpeas, the library we use for enabling applets / extensions, does not support bash scripting, this is a scripting language intended for command interpreters and is not an appropriate language for the usecase of applets. I would recommend learning Python or Vala. You can write an applet in C, Python, or Vala.

**Comment by lazylavenders** [comment]
1. Having options for accent colour like the recent GNOME 42? I found that feature to be great on gnome.

2. Ability to save custom panel layouts. It was kind of annoying for me to reconfigure the panel because after most of the updates the panel would disappear but with the budgie-panel's reset command all the settings would be gone.

3. Ability to create custom applets for Raven. I would love to create a news feed applet for the raven panel. 

Thats all for now, but I would be extremely extremely extremely glad if we got blur features like KDE. Good luck 🤞

**Reply by rbreaves** [reply]
Blur has been tied w/ mutter & the gnome shell so not happening on 10.x. Who knows on 11 though w/ FTL or whatever it’s called.

Also blur effects in general prevent GPUs of all kinds, low to high end from being able to power down into low wattage states so while it looks great you’re also needlessly burning the battery life. Fine for a desktop but maybe not for a laptop.

**Comment by Solarunit** [comment]
What about built-in true OLED dark theme optimized for the OLED screens? :bulb: 
Because more and more screens are OLED now.

**Comment by Solarunit** [comment]
Could we have configurable `Alt + F2` Launcher shortcut? Alt + F2 not an "easy shortcut" compared to single win/super.

I am thinking about `Win` / `Super` as an option,  sure we need to replace Munu  shortcut in this case.
Another option is `Shift + Win` / `Ctrl + Win` ?

**Reply by rbreaves** [reply]
I think in both cases you are talking about things you can customize yourself or is an edge case. OLED really is not that common imo and changing default hotkeys just seems like an odd request. As far as I knew the Super key does launch the menu either way I change most of the system level hotkeys with kinto.sh in one fell swoop.

**Reply by JoshStrobl** [reply]
@Solarunit True story, I added Super as a secret valid launch key for the Alt+F2 dialog. I don't want to promote it as a stable and final option since I won't have it in 11, but you can use it in 10.x already by doing the following:

```
gsettings set com.solus-project.budgie-wm experimental-enable-run-dialog-as-menu true
```

You can already configure Budgie Run Dialog to launch with another shortcut though. Open up Budgie Control Center, go to Keyboard and then view the shortcuts at the bottom. Under System, change the "Show the run command prompt". I can't guarantee that everything will work, nor is that really something I care to broadly support, but at the very least I just set it to Win+R as a test and it worked as expected

**Reply by Solarunit** [reply]
ok, I enabled super to launch the Alt+F2 dialog, will use it for a while. But at the same time I would like to use something like Super + Shift for the Budgie menu or vice versa.

BTW looks like you can't assign  Super + Shift or anything with Super in the Budgie Control Center.
Also search (sorting) in Alt+F2 dialog works pretty bad atm.
What about to add some features to it (like restart, shutdown etc) and make it as powerful as  Vivaldi F2 Menu? :bulb:

**Reply by Solarunit** [reply]
>OLED really is not that common imo

not that common yet. And not only OLED displays but also Mini-LED become more popular, so true dark theme is a Must-have imo.

**Reply by JoshStrobl** [reply]
> What about to add some features to it (like restart, shutdown etc) and make it as powerful as Vivaldi F2 Menu? bulb

It is the run dialog. It is meant to just quickly launch apps, and nothing else.

> Also search (sorting) in Alt+F2 dialog works pretty bad atm.

Already worked on in a Budgie Menu rewrite PR.

**Comment by Solarunit** [comment]
It was discussed earlier about Multiple monitor setups with virtual desktops separation. 
I have another idea: so I would like to use 2nd screen as a Standalone Monitor and disable mouse transition between them. And use some kind of modifier keys (Ctrl) to go to the another monitor or drag an app. Or maybe some kind of applet to quickly switch between monitors...

**Reply by rbreaves** [reply]
Can probably do that with x11 already, and some scripting to jump the mouse btwn them. If displays aren't butting up against each other then yea.. the mouse shouldn't travel btwn them. Bigger issue is getting your budgie panel to follow you or duplicate.

**Reply by Solarunit** [reply]
I found **Separate screens** section in the https://wiki.archlinux.org/title/multihead
Looks like this is what i need.

>**Separate screens**
>This is the original way of configuring multiple monitors with X, and it has been around for decades. Each physical monitor is assigned as an X screen, and while you can move the mouse between them, they are more or less independent.

>Normally the X display has a single identifier such as :0, set in the DISPLAY environment variable; but in this configuration, each screen has a different $DISPLAY value. The first screen is :0.0, the second is :0.1 and so on.

>With this configuration, it is not possible to move windows between screens, apart from a few special programs like GIMP and Emacs, which have multi-screen support. For most programs, you must change the DISPLAY environment variable when launching to have the program appear on another screen:

and some xorg.conf examples here https://unix.stackexchange.com/questions/365796/xorg-independent-mode

Tried it with no success. 
If you have this setup please share (I am on Solus Budgie)

**Reply by JoshStrobl** [reply]
This post isn't intended to be for support requests. Let's keep things on-topic.

**Reply by rbreaves** [reply]
What about just use 3 user accounts? A viewer user account, and then your 2 actual user accounts that you would access over x11vnc or whatever vnc server you install? Could easily maximize each of those to both monitors, no spanning issue and no issue with lacking a budgie-panel on one monitor or another.

Edit: Ok I am done, just seeing Josh's response. But I get the feeling virtual desktop separation isn't likely to be implemented, practically no DE's currently do this and even Windows can't figure it out. Only macOS has it working right imo.

**Comment by Solarunit** [comment]
Another idea: new Screensaver mode: Dim (like Kodi has)

with settings:

- wait time
- brightness level: 0-100%

**Reply by JoshStrobl** [reply]
Not really in scope for Budgie 10 as this would be currently exposed and defined through gnome-settings-daemon as opposed to our own.
